weave-js/src/common/hooks/useLifecycleProfiling.ts

```diff
@@ -14,11 +14,16 @@ export type ProfileData = {
  */
 export function useLifecycleProfiling(
   id: string,
-  cb: (d: ProfileData) => void
+  cb: (d: ProfileData) => void,
+  onStart?: (name: string) => void
 ) {
   useEffect(() => {
     const x = performance.now();
 
+    if (onStart) {
+      onStart(id);
+    }
+
```

Comment:
this can be simplified to `onStart?.(id)`

Reply:
right... Why did I not do that? 😠